### PR TITLE
[test] Porting tests to new syntax

### LINKED ISSUE #592: Adapt the remaining tests to the new syntax

### PR DESCRIPTION
**Pull request by kraushm**
This is a first set of remaining tests being ported to the new syntax.

Fixes #592.

### CODE REVIEW

File: cscs-checks/apps/amber/amber_check.py [hunk 1]
```diff
@@ -1,22 +1,28 @@
...
-        self.executable_opts = \
-            ('-O -i %s -o %s' % (input_file, output_file)).split()
+
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.modules = ['Amber']
+        self.num_tasks = 1
+        self.num_tasks_per_node = 1
+        self.num_gpus_per_node = 1
+        self.executable_opts = ('-O -i %s -o %s'
+                                % (input_file, output_file)).split()
```
[the remark this answers is not in the document]
Comment (kraushm) [reply cscs-checks/apps/amber/amber_check.py]: Changed according to review.

File: cscs-checks/apps/amber/amber_check.py [hunk 2]
```diff
@@ -31,84 +37,53 @@ def __init__(self, name, input_file, output_file, **kwargs):
...
-        else:
-            self.num_tasks = 576
-            self.num_tasks_per_node = 36
-
+@rfm.required_version('>=2.14')
+@rfm.parameterized_test(*([variant, architecture]
+                          for variant in ['prod', 'maint']
+                          for architecture in ['CPU', 'GPU']))
+class AmberCheck(AmberBaseCheck):
+    def __init__(self, variant, architecture):
```
[the remark this answers is not in the document]
Comment (kraushm) [reply cscs-checks/apps/amber/amber_check.py]: Changed according to review.

File: cscs-checks/apps/espresso/espresso_check.py [hunk 3]
```diff
@@ -1,88 +1,81 @@
...
+import reframe as rfm
 import reframe.utility.sanity as sn
-from reframe.core.pipeline import RunOnlyRegressionTest
 
 
-class EspressoBaseCheck(RunOnlyRegressionTest):
-    def __init__(self, variant, **kwargs):
-        super().__init__('quantum_espresso_%s_check' % variant,
-                         os.path.dirname(__file__), **kwargs)
+@rfm.required_version('<=2.14')
```
[the remark this answers is not in the document]
Comment (kraushm) [reply cscs-checks/apps/espresso/espresso_check.py]: This was probably due to copy and paste and was removed.

File: cscs-checks/apps/espresso/espresso_check.py [hunk 4]
```diff
@@ -1,88 +1,81 @@
...
+class EspressoBaseCheck(rfm.RunOnlyRegressionTest):
```
[the remark this answers is not in the document]
Comment (kraushm) [reply cscs-checks/apps/espresso/espresso_check.py]: Changed according to review.

File: cscs-checks/apps/espresso/espresso_check.py [hunk 5]
```diff
@@ -1,88 +1,81 @@
...
         }
 
 
-class EspressoGPUCheck(EspressoBaseCheck):
-    def __init__(self, **kwargs):
-        super().__init__('gpu', **kwargs)
-
-        self.executable = 'pw-gpu.x'
-        self.valid_systems = ['daint:gpu', 'dom:gpu']
+@rfm.parameterized_test(['cpu'], ['gpu'])
```
[the remark this answers is not in the document]
Comment (kraushm) [reply cscs-checks/apps/espresso/espresso_check.py]: GPU version has been removed.

File: cscs-checks/apps/espresso/espresso_check.py [hunk 6]
```diff
@@ -1,88 +1,81 @@
...
+class EspressoProdCheck(EspressoBaseCheck):
```
[the remark this answers is not in the document]
Comment (kraushm) [reply cscs-checks/apps/espresso/espresso_check.py]: `class EspressoProdCheck` was merged into single check.

File: cscs-checks/cuda/hpcg_check.py [hunk 7]
```diff
@@ -1,24 +1,21 @@
...
 import reframe.utility.sanity as sn
-from reframe.core.pipeline import RunOnlyRegressionTest, RegressionTest
 
 
-class HPCGCheck(RunOnlyRegressionTest):
-    def __init__(self, **kwargs):
-        super().__init__('hpcg_check',
-                         os.path.dirname(__file__), **kwargs)
+@rfm.simple_test
+class HPCGCheck(rfm.RunOnlyRegressionTest):
```
[the remark this answers is not in the document]
Comment (kraushm) [reply cscs-checks/cuda/hpcg_check.py]: Class was renamed and test was moved to microbenchmarks.

File: cscs-checks/apps/amber/amber_check.py [hunk 8]
```diff
@@ -1,22 +1,27 @@
 import os
-
+import reframe as rfm
```
Comment (teojgo) [inline cscs-checks/apps/amber/amber_check.py]: Put an empty line here to distinguish between the standard library modules and the ReFrame ones.

File: cscs-checks/apps/amber/amber_check.py [hunk 9]
```diff
@@ -31,84 +36,53 @@ def __init__(self, name, input_file, output_file, **kwargs):
...
-            self.num_tasks_per_node = 36
-
+@rfm.required_version('>=2.14')
+@rfm.parameterized_test(*([variant, arch]
+                          for variant in ['prod', 'maint']
+                          for arch in ['CPU', 'GPU']))
+class AmberCheck(AmberBaseCheck):
+    def __init__(self, variant, arch):
+        super().__init__('mdin.%s' % arch, 'amber.out')
+        if arch== 'GPU':
```
Comment (teojgo) [inline cscs-checks/apps/amber/amber_check.py]: Add the whitespace missing after `arch`

File: cscs-checks/apps/amber/amber_check.py [hunk 10]
```diff
@@ -31,84 +36,53 @@ def __init__(self, name, input_file, output_file, **kwargs):
...
+                    'perf': (30.0, -0.05, None)
+                },
+            }
+            if variant == 'prod':
+                self.descr = 'Amber parallel GPU production check'
+                self.tags |= {'production'}
+            elif variant == 'maint':
+                self.descr = 'Amber parallel GPU maintenance check'
+                self.tags |= {'maintenance'}
+        elif arch== 'CPU':
```
Comment (teojgo) [inline cscs-checks/apps/amber/amber_check.py]: Add the whitespace missing after `arch`

File: cscs-checks/apps/amber/amber_check.py [hunk 11]
```diff
@@ -31,84 +36,53 @@ def __init__(self, name, input_file, output_file, **kwargs):
...
+                self.descr = 'Amber parallel CPU production check'
+                self.tags |= {'production'}
+                self.executable = 'pmemd.MPI'
+                self.strict_check = False
+                if self.current_system.name == 'dom':
+                    self.num_tasks = 216
+                    self.num_tasks_per_node = 36
+                else:
+                    self.num_tasks = 576
+                    self.num_tasks_per_node = 36
```
Comment (teojgo) [inline cscs-checks/apps/amber/amber_check.py]: Put an empty line after the end of the else block

File: cscs-checks/apps/cpmd/cpmd_check.py [hunk 12]
```diff
@@ -1,24 +1,36 @@
 import os
-
+import reframe as rfm
```
Comment (teojgo) [inline cscs-checks/apps/cpmd/cpmd_check.py]: Add empty line to distinguish the standard library/ReFrame imports

File: cscs-checks/apps/cpmd/cpmd_check.py [hunk 13]
```diff
@@ -1,24 +1,36 @@
...
         self.num_tasks_per_node = 1
         self.use_multithreading = True
+        self.strict_check = False
+        self.extra_resources = {
+            'switches': {
+                'num_switches': 1
+            }
+        }
+        #  OpenMP version of CPMD segfaults
+        #  self.variables = { 'OMP_NUM_THREADS' : '8' }
```
Comment (teojgo) [inline cscs-checks/apps/cpmd/cpmd_check.py]: Remove empty line

File: cscs-checks/apps/espresso/espresso_check.py [hunk 14]
```diff
@@ -1,52 +1,48 @@
 import os
-
+import reframe as rfm
```
Comment (teojgo) [inline cscs-checks/apps/espresso/espresso_check.py]: As above

File: cscs-checks/apps/espresso/espresso_check.py [hunk 15]
```diff
@@ -1,52 +1,48 @@
...
-        super().__init__('quantum_espresso_%s_check' % variant,
-                         os.path.dirname(__file__), **kwargs)
-        self.descr = 'Quantum Espresso check (%s)' % variant
-        self.valid_prog_environs = ['PrgEnv-intel']
-        self.modules = ['QuantumESPRESSO']
+@rfm.simple_test
+class QECheck(rfm.RunOnlyRegressionTest):
+    def __init__(self):
+        super().__init__()
+
```
Comment (teojgo) [inline cscs-checks/apps/espresso/espresso_check.py]: Remove this empty line

File: cscs-checks/apps/espresso/espresso_check.py [hunk 16]
```diff
@@ -1,52 +1,48 @@
...
-                                    self.stdout, 'sec', float)
-        }
-        self.use_multithreading = True
+        self.executable_opts = ['-in', 'ausurf.in']
         if self.current_system.name == 'dom':
             self.num_tasks = 216
             self.num_tasks_per_node = 36
         else:
             self.num_tasks = 576
             self.num_tasks_per_node = 36
```
Comment (teojgo) [inline cscs-checks/apps/espresso/espresso_check.py]: Add empty line after else block.

File: cscs-checks/microbenchmarks/hpcg/hpcg_check.py [hunk 17]
```diff
@@ -1,24 +1,21 @@
 import os
-
+import reframe as rfm
```
Comment (teojgo) [inline cscs-checks/microbenchmarks/hpcg/hpcg_check.py]: As above

File: cscs-checks/microbenchmarks/hpcg/hpcg_check.py [hunk 18]
```diff
@@ -1,24 +1,21 @@
...
 
 
-class HPCGCheck(RunOnlyRegressionTest):
-    def __init__(self, **kwargs):
-        super().__init__('hpcg_check',
-                         os.path.dirname(__file__), **kwargs)
+@rfm.simple_test
+class HPCG_GPUCheck(rfm.RunOnlyRegressionTest):
+    def __init__(self):
+        super().__init__()
```
Comment (teojgo) [inline cscs-checks/microbenchmarks/hpcg/hpcg_check.py]: Remove empty line

File: cscs-checks/prgenv/opencl_check.py [hunk 19]
```diff
@@ -1,26 +1,20 @@
 import os
-
+import reframe as rfm
```
Comment (teojgo) [inline cscs-checks/prgenv/opencl_check.py]: As above

File: cscs-checks/prgenv/opencl_check.py [hunk 20]
```diff
@@ -1,26 +1,20 @@
...
+        self.tags = {'production'}
 
-class OpenCLCheck(RegressionTest):
-    def __init__(self, **kwargs):
-        super().__init__('opencl_check',
-                         os.path.dirname(__file__), **kwargs)
         self.valid_systems = ['daint:gpu', 'dom:gpu']
         self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-pgi']
-
         self.modules = ['cudatoolkit']
```
Comment (teojgo) [inline cscs-checks/prgenv/opencl_check.py]: Use craype-accel-nvidia60

File: cscs-checks/microbenchmarks/hpcg/hpcg_check.py [hunk 21]
```diff
@@ -1,24 +1,21 @@
...
 
+        self.maintainers = ['VK']
         self.descr = 'HPCG check'
+        self.sourcesdir = os.path.join(self.current_system.resourcesdir, 'HPCG')
+
         self.valid_systems = ['daint:gpu']
         self.valid_prog_environs = ['PrgEnv-gnu']
         self.modules = ['cudatoolkit', 'craype-hugepages8M']
```
Comment (teojgo) [inline cscs-checks/microbenchmarks/hpcg/hpcg_check.py]: Use craype-accel-nvidia60

File: cscs-checks/apps/amber/amber_check.py (hunk 9 above)
Comment (kraushm) [reply cscs-checks/apps/amber/amber_check.py]: Changed according to review.

File: cscs-checks/apps/amber/amber_check.py (hunk 10 above)
Comment (kraushm) [reply cscs-checks/apps/amber/amber_check.py]: Changed according to review.

File: cscs-checks/microbenchmarks/hpcg/hpcg_check.py (hunk 18 above)
Comment (kraushm) [reply cscs-checks/microbenchmarks/hpcg/hpcg_check.py]: Changed according to review.

File: cscs-checks/prgenv/opencl_check.py (hunk 20 above)
Comment (kraushm) [reply cscs-checks/prgenv/opencl_check.py]: Changed according to review.

File: cscs-checks/microbenchmarks/hpcg/hpcg_check.py (hunk 21 above)
Comment (kraushm) [reply cscs-checks/microbenchmarks/hpcg/hpcg_check.py]: Changed according to review.

File: cscs-checks/apps/cpmd/cpmd_check.py (hunk 13 above)
Comment (kraushm) [reply cscs-checks/apps/cpmd/cpmd_check.py]: Changed according to review.

File: cscs-checks/apps/espresso/espresso_check.py (hunk 15 above)
Comment (kraushm) [reply cscs-checks/apps/espresso/espresso_check.py]: Changed according to review.

**Comment by vkarak** [comment]
@jenkins-cscs retry all

**Comment by vkarak** [comment]
@victorusu @teojgo With this PR the name of the QE test changes. This will affect your scripts. Are you ok with that?

**Comment by teojgo** [comment]
@vkarak I can easily adapt to take into account the name change.

**Comment by victorusu** [comment]
@vkarak and @kraushm, I am ok with that. I will only have to adapt two scripts.

**Comment by vkarak** [comment]
@jenkins-cscs retry none

**Pull request merged**